pkg/controller/backup_storage_location_controller.go

```diff
@@ -53,96 +60,88 @@ type BackupStorageLocationReconciler struct {
 // +kubebuilder:rbac:groups=velero.io,resources=backupstoragelocations,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=velero.io,resources=backupstoragelocations/status,verbs=get;update;patch
 func (r *BackupStorageLocationReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	log := r.Log.WithField("controller", BackupStorageLocation)
+	var unavailableErrors []string
+	var location velerov1api.BackupStorageLocation
 
-	log.Debug("Validating availability of backup storage locations.")
+	log := r.Log.WithField("controller", BackupStorageLocation).WithField(BackupStorageLocation, req.NamespacedName.String())
+	log.Debug("Validating availability of BackupStorageLocation")
 
 	locationList, err := storage.ListBackupStorageLocations(r.Ctx, r.Client, req.Namespace)
 	if err != nil {
-		log.WithError(err).Error("No backup storage locations found, at least one is required")
+		log.WithError(err).Error("No BackupStorageLocations found, at least one is required")
```

Comment:
I don't think you need to return the error in L72.

In a similar case like following, the error is not returned:
```
	if location.Name == "" || location.Namespace == "" {
		log.WithError(err).Error("BackupStorageLocation is not found")
		return ctrl.Result{}, nil
	}
```

Please also check the behavior of the controller if an error is returned, I thought it would trigger the `Reconcile` again.

Reply:
Checked controller-runtime code, when reconcile function return error, the req key will be re-enter queue after that:
https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/internal/controller/controller.go#L322-L326 

``` go
	result, err := c.Reconcile(ctx, req)
	switch {
	case err != nil:
		c.Queue.AddRateLimited(req)
		ctrlmetrics.ReconcileErrors.WithLabelValues(c.Name).Inc()
		ctrlmetrics.ReconcileTotal.WithLabelValues(c.Name, labelError).Inc()
		log.Error(err, "Reconciler error")
```